### PR TITLE
Add HAVT I2C Responses

### PR DESCRIPTION
**Pull request by shihaocao**
This small PR merely makes the write and read registers available within ADCS Software. It does not yet actually "do" anything with them.

Here is the sister PR from FlightSoftware:
https://github.com/pathfinder-for-autonomous-navigation/FlightSoftware/pull/135

### CODE REVIEW

File: src/state.cpp [hunk 1]
```diff
@@ -1,62 +1,67 @@
...
+    0.0f,               // TODO : Gyroscope temperature K_i
+    0.0f,               // TODO : Gyroscope temperature K_d
+    0.0f,               // TODO : Gyroscope desired temperature
+  },
+  // HAVT Registers
+  {
+    0xF0F01F1F, // Read register
+    0, // Cmd register
+    0, // Cmd flag
+  }
```
Comment (shihaocao) [inline src/state.cpp]: This section and the speed section are the only updated ones

File: src/state.cpp [hunk 2]
```diff
@@ -1,62 +1,67 @@
...
+  0,    // ADCS Mode
+  0,    // Read Pointer
+  // Reaction Wheel Registers
+  {
+    0,                  // Mode
+    {0.0f, 0.0f, 0.0f}, // Command
+    0,                  // Command Flag
+    1.0f,               // Momentum filter
+    1.0f,               // Ramp filter
+    {400.0f, -300.0f, 200.0f}, // Speed read
```
Comment (shihaocao) [inline src/state.cpp]: Updated this line and the section below, all else shouldn't have changed

File: include/state.hpp [hunk 3]
```diff
@@ -1,134 +1,148 @@
...
+  /** This table is updated with the most recent is_functional() for
+   * each box aboard the ADCS Box. */
+  unsigned int read_table;
+  /** An intermediate data structure that will specify the requested reset() and disable()
+   * for devices within the ADCS Box. */
+  unsigned int cmd_table;
+  /** This flag specifies if there is a new_flag, and is set to outdated
+   * once the cmd_table has been applied. */
+  unsigned char cmd_flag;
+};
```
Comment (shihaocao) [inline include/state.hpp]: This is the only new block of code

File: include/state.hpp [hunk 4]
```diff
@@ -1,134 +1,148 @@
...
+  struct ReactionWheelRegisters rwa;
+  /** Magnetorquer state struct. */
+  struct MagnetorquerRegisters mtr;
+  /** Sun sensor state struct. */
+  struct SunSensorRegisters ssa;
+  /** IMU state struct. */
+  struct IMURegisters imu;
+  /** Hardware availability table bit string. */
+  struct HAVTRegisters havt;
+};
```
Comment (shihaocao) [inline include/state.hpp]: and here

File: src/state.cpp (hunk 2 above)
Comment (tanishqaggarwal) [reply src/state.cpp]: Why was speed updated?

File: src/state.cpp (hunk 1 above)
Comment (tanishqaggarwal) [reply src/state.cpp]: What's the significance of 0xF0F01F1F?

File: src/state.cpp (hunk 2 above)
Comment (tanishqaggarwal) [reply src/state.cpp]: Oh because of speed -> momentum, got it

File: src/state.cpp (hunk 1 above)
Comment (shihaocao) [reply src/state.cpp]: Absolutely nothing. It was just unique enough for me to test endianness.

**Pull request merged**